### PR TITLE
Some adjustments for Storage Driver and SourceManager to run on the cloud

### PR DESCRIPTION
**Pull request by andresgutgon**
## Describe your changes
We want to use some methods from these 2 packages in our Latitude cloud and we have to make it work.

# TODO
- [x] Do specs for resolve secrets function

### CODE REVIEW

File: packages/source_manager/src/manager/index.ts [hunk 1]
```diff
@@ -201,7 +202,9 @@ export default class SourceManager {
       const startTime = performance.now()
       const compiled = await source.compileQuery({ queryPath, params: {} })
 
-      const stream = await this.materializedStorage.createWriteStream(filename)
+      const stream = (await this.materializedStorage.createWriteStream(
+        filename,
+      )) as Writable
```
Comment (andresgutgon) [inline packages/source_manager/src/manager/index.ts:207]: I don't understand why `materializedStorage` is typed a `any`. We must be doing something wrong.

File: packages/storage_driver/src/drivers/base.ts [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
 
 export abstract class StorageDriver {
   constructor() {}
 
+  public abstract type: StorageType
+
+  get isS3Driver(): boolean {
+    return this.type === StorageType.s3
+  }
+
```
Comment (csansoon) [inline packages/storage_driver/src/drivers/base.ts:12]: What is this for? Can't you just compare the type?

Comment (andresgutgon) [reply packages/storage_driver/src/drivers/base.ts:12]: I can, but this is nicer

Comment (csansoon) [reply packages/storage_driver/src/drivers/base.ts:12]: I think the an abstract parent class like this should be agnostic to its children subclasses existence.
This is not as big of a deal as if it was on the BaseConnector though, here's only two subclasses (for now). Just OCD.

**Review by csansoon — APPROVED** [review]
Bit of OCD. Everything else is nice 👌

File: packages/storage_driver/src/drivers/base.ts (hunk 2 above)
Comment (andresgutgon) [reply packages/storage_driver/src/drivers/base.ts:12]: Yep pure ocd xD

**Pull request merged**